graviti/portex/factory.py

```diff
@@ -43,6 +46,134 @@ def __call__(self, **_: Any) -> Any:
         ...
 
 
+class ImmutableFieldsFactory:
+    """The factory for ImmutableFields.
+
+    Arguments:
+        decl: The decalaration of immutable fields.
+        imports: The :class:`Imports` instance to specify the import scope of the fields.
+
+    """
+
+    def __init__(self, decl: Iterable[Dict[str, Any]], imports: Imports) -> None:
+        self._factories = [FieldFactory(field, imports) for field in decl]
+
+    def __call__(self, **kwargs: Any) -> ImmutableFields:
+        """Apply the input arguments to the ImmutableFields factory.
+
+        Arguments:
+            kwargs: The input arguments.
+
+        Returns:
+            The applied ImmutableFields.
+
+        """
+        return ImmutableFields(
+            filter(
+                bool,
+                (factory(**kwargs) for factory in self._factories),  # type: ignore[misc]
+            )
+        )
+
+
+class BaseImmutableFieldsFactory:
+    """The factory for ConnectedFields.
+
+    Arguments:
+        factory: The factory of immutable fields.
+        internal_kwargs_getter: The method to get the kwargs of the base type.
+
+    """
+
+    def __init__(
+        self,
+        factory: Callable[..., UnionFields],
```

Comment:
should the type of factory be `Union[BaseImmutableFieldsFactory, ImmutableFieldsFactory]`